pkg/fatfs/fatfs_vfs/fatfs_vfs.c

```diff
@@ -122,6 +122,31 @@ static int _umount(vfs_mount_t *mountp)
     return fatfs_err_to_errno(res);
 }
 
+static int _statvfs(vfs_mount_t *mountp, const char *restrict path, struct statvfs *restrict buf)
+{
+    (void)path;
+    fatfs_desc_t *fs_desc = (fatfs_desc_t *)mountp->private_data;
+    mtd_dev_t *mtd = fs_desc->dev;
+    DWORD nclst;
+    FATFS *fs;
+
+    int res = f_getfree(fs_desc->abs_path_str_buff, &nclst, &fs);
+    if (res != FR_OK) {
+        return fatfs_err_to_errno(res);
+    }
+
+    unsigned sector_size = mtd->page_size * mtd->pages_per_sector;
+
+    buf->f_bsize  = fs->csize * sector_size;
```

Comment:
There are uninitialized fields. That's consistent with some other FSs (eg. both littlefs; OTOH, spiffs and constfs do zero), but neither `vfs_statvfs` nor the typical user (`_print_df` in `sc_vfs.c`) zeroes that buffer, and as there is no clear specification in RIOT as to which fields are valid to read and which not, a zero memset might be starting a good practice here.

(I tried adding a suggestion, but for some reason that seems not to work here.)

Reply:
Actually, as only littlefs does this (and looking through online examples shows that no caller initializes their buf when doing this on POSIX either), I consider it a bug and propose https://github.com/RIOT-OS/RIOT/pull/17644 to fix it; thus altering my review from ACK to COMMENT until this is addressed here too.

Reply:
IMHO it should be the responsibility of the VFS layer to zero the buffer, this will always ensure a consistent state for all implementations